pkg/cableengine/syncgatewaystatus.go

```diff
@@ -0,0 +1,161 @@
+package cableengine
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"strconv"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+
+	v1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/log"
+)
+
+const GatewayUpdateIntervalSeconds = 5
+const GatewayStaleTimeoutSeconds = GatewayUpdateIntervalSeconds * 3
+const updateTimestampAnnotation = "update-timestamp"
+
+func (i *engine) syncGatewayStatus() {
+
+	i.Lock()
+	defer i.Unlock()
+
+	klog.V(log.TRACE).Info("Running syncGatewayStatus()")
+
+	gatewayObj, err := i.generateGatewayObject()
+	if err != nil {
+		klog.Errorf("generating gateway object from driver connections: %s", err)
+		return
+	}
+
+	existingGw, err := i.getLastSyncedGateway(gatewayObj.Name)
+
+	// log and stop for any error different to a not-found error
+	if err != nil && !errors.IsNotFound(err) {
+		klog.Errorf("trying to read existing gateway from k8s api: %s", err)
+		return
+	}
+
+	gwClient := i.clientset.SubmarinerV1().Gateways(i.namespace)
+
+	if err != nil && errors.IsNotFound(err) {
+		klog.V(log.TRACE).Infof("Gateway object needs creation: %+v", gatewayObj)
+		_, err = gwClient.Create(gatewayObj)
+		if err != nil {
+			klog.Errorf("creating Gateway object: %s", err)
+			return
+		}
+	} else {
+		if !reflect.DeepEqual(gatewayObj.Status, existingGw.Status) {
+			klog.V(log.TRACE).Infof("Gateway object needs an update: %+v", gatewayObj)
+			existingGw.Status = gatewayObj.Status
+			existingGw.Labels = gatewayObj.Labels
+			existingGw.Annotations = gatewayObj.Annotations
+
+			gw, err := gwClient.Update(existingGw)
+			if err != nil {
+				klog.Errorf("updating Gateway object: %s", err)
+				return
+			} else {
+				klog.V(log.TRACE).Infof("Gateway updated correctly: %+v", gw)
+			}
```

Comment:
Shall we use retry.RetryOnConflict here?

Reply:
It's supposed to be only one updater per gateway, a conflict would be indicative of something very very wrong in the system (2 gateways running on the same host?). IMO I think it's better not to retry.